.dev/lint_metadata_test.R

```diff
@@ -0,0 +1,45 @@
+# This script is designed to find linters that lack metadata tests.
+#   To do so, it forces xml_nodes_to_lints() to give the wrong information,
+#   runs the test suite, and finds linters that nevertheless pass all their tests.
+library(testthat)
+
+xml_nodes_to_lints_file <- "R/xml_nodes_to_lints.R"
+
+xml_nodes_to_lints_file |>
+  readLines() |>
+  sub(
+    pattern = "line_number = as.integer(line1)",
+    replacement = "line_number = as.integer(2^31 - 1)"
+  ) |>
+  writeLines(xml_nodes_to_lints_file)
+
+pkgload::load_all()
+
+report <- test_dir(
+  "tests/testthat",
+  filter = "linter$",
+  stop_on_failure = FALSE,
+  reporter = SilentReporter$new()
+)
+names(report) <- vapply(report, `[[`, "file", FUN.VALUE = character(1L))
+
+# Hack the nested structure of the testthat report to identify which files have
+#   any failed test
+failed <- report |>
+  vapply(
+    \(x) any(vapply(x$results, inherits, "expectation_failure", FUN.VALUE = logical(1L))),
+    logical(1L)
+  ) |>
+  which() |>
+  names() |>
+  unique() |>
+  gsub(pattern = "^test-|\\.R$", replacement = "")
+
+passed <- setdiff(
+  available_linters(tags = NULL)$linter,
+  failed
+)
+
+if (length(passed) > 0L) {
+  stop("Please add tests of lint metadata for the following linters: ", toString(passed))
```

Comment:
Can we display one linter per line here? Will make the output easier to peruse.

Reply:
We could, but it's only really an issue for this initial PR. Later, we'd only see one or two files affected once this first batch is cleared. So I'd leave it as is for simplicity.

Reply:
Good point.